[Asset Inventory] Filter out empty entity.id documents

opauloh opened the pull request:
## Summary

This closes #224982

Asset Inventory relies on the `entity.id` field as a unique identifier for visualizations, grouping, filtering, and flyout functionality. Currently, documents missing this field are included in the results, leading to noise, broken interactions, and misleading asset entries.

This PR implements filtering within the Asset Inventory fetchers and UI components to ensure that only valid documents with `entity.id` are processed and displayed.


### Screenshot



**Before**
Unexpected behaviour in the Inventory page due to empty `user.name`

<img width="2162" alt="Image" src="https://github.com/user-attachments/assets/18131bfc-c05e-4165-ab86-fea03b0a1c49" />

**After:**


<img width="1985" alt="image" src="https://github.com/user-attachments/assets/0a3ca7de-b237-4d97-b62c-6fbd665e8cc5" />

opauloh committed:
Adds util to filter empty entity IDs

Adds a utility function to filter out assets with empty entity IDs.

This ensures that only assets with valid entity IDs are displayed,
improving data quality and preventing potential errors.
The utility provides functions that can work on filter arrays
or query bool filter arrays.

opauloh committed:
Enhances asset inventory filtering

Improves asset inventory filter functionality by ensuring filters are correctly applied based on the current data view.

This change adds a loading state to the search bar and ensures a default filter is created to prevent errors when no data is available for the specified data view. It also passes the filters and query to the `FilterGroup` component.

opauloh committed:
Improves asset inventory data filtering

Ensures that queries for asset inventory data exclude entries with empty or missing data. This prevents incomplete records from being displayed, leading to a more accurate and reliable representation of assets.

Adds a loading state to the search bar.

elasticmachine commented:
Pinging @elastic/kibana-cloud-security-posture (Team:Cloud Security)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311811)
* Commit: c33cbf91c550a40187b0e79b727aecfb4a6736a9

### Failed CI Steps
* [Jest Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/311811#01979fa5-acd4-4800-88df-df4b969be953)
* [Jest Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/311811#01979fc7-3e12-42a9-8517-d4096e869574)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311811#01979fa5-acd4-4800-88df-df4b969be953) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311811/jobs/01979fa5-acd4-4800-88df-df4b969be953/artifacts/01979fc7-256f-443f-a553-a21baa54757c) Jest Tests #<span></span>10 / getTopAssetsQuery should handle missing query subfields safely
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311811#01979fc7-3e12-42a9-8517-d4096e869574) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311811/jobs/01979fc7-3e12-42a9-8517-d4096e869574/artifacts/01979fcc-575a-4b19-8c8e-1f8f42db202d) Jest Tests #<span></span>10 / getTopAssetsQuery should handle missing query subfields safely
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311811#01979fa5-acd4-4800-88df-df4b969be953) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311811/jobs/01979fa5-acd4-4800-88df-df4b969be953/artifacts/01979fc7-2570-435a-91be-307337d7006c) Jest Tests #<span></span>10 / getTopAssetsQuery should return a valid ES query with aggregations by entityType and entitySubType using entityId count
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311811#01979fc7-3e12-42a9-8517-d4096e869574) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311811/jobs/01979fc7-3e12-42a9-8517-d4096e869574/artifacts/01979fcc-575a-4023-b251-b0c6c5dbc347) Jest Tests #<span></span>10 / getTopAssetsQuery should return a valid ES query with aggregations by entityType and entitySubType using entityId count

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311811","number":311811,"commit":"c33cbf91c550a40187b0e79b727aecfb4a6736a9"}],"number":311811}
buildkite-pr-comment-->

opauloh committed:
Adds filter for empty entity IDs in asset queries

Ensures that queries for top assets exclude results with empty entity IDs.

This prevents displaying incomplete or invalid asset data in charts
and tables by filtering out assets that are missing essential
identification information.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311832)
* Commit: d59eab2a9cc582fc9eb956f48b4628441e34a497

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/4553774d5a3ed332be03eca6474208f157fd5b83) | [after](https://github.com/elastic/kibana/commit/d59eab2a9cc582fc9eb956f48b4628441e34a497) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7674 | 7675 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/4553774d5a3ed332be03eca6474208f157fd5b83) | [after](https://github.com/elastic/kibana/commit/d59eab2a9cc582fc9eb956f48b4628441e34a497) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.4MB | 9.4MB | +412.0B |



### History
* :broken_heart: [Build #311811](https://buildkite.com/elastic/kibana-pull-request/builds/311811) failed c33cbf91c550a40187b0e79b727aecfb4a6736a9

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311832","number":311832,"commit":"d59eab2a9cc582fc9eb956f48b4628441e34a497"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311811","number":311811,"commit":"c33cbf91c550a40187b0e79b727aecfb4a6736a9"}],"number":311832}
buildkite-pr-comment-->
